tests/unit/igraph_get_k_shortest_paths.c

```diff
@@ -0,0 +1,112 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+#include "test_utilities.h"
+
+void call_and_print(
+    igraph_t *graph, igraph_vector_t *weights, igraph_integer_t k,
+    igraph_integer_t from, igraph_integer_t to, igraph_neimode_t mode
+) {
+    igraph_vector_int_list_t paths;
+
+    igraph_vector_int_list_init(&paths, 0);
+    IGRAPH_ASSERT(igraph_get_k_shortest_paths(graph, weights, &paths, k, from, to, mode) == IGRAPH_SUCCESS);
+    printf("result: \n");
+    print_vector_int_list(&paths);
+    igraph_vector_int_list_destroy(&paths);
+    printf("\n");
+
+    VERIFY_FINALLY_STACK();
+}
+
+
+int main() {
+    /* Wiki example taken from https://en.wikipedia.org/wiki/Yen's_algorithm */
+    igraph_t g_0, g_1, g_2, g_2c, g_wiki, g_wiki_u;
+    igraph_vector_t weights, weights_wiki, weights_inf;
+    igraph_vector_int_list_t paths;
+
+    igraph_vector_int_list_init(&paths, 0);
+
+    igraph_small(&g_0, 0, 0, -1);
+    igraph_small(&g_1, 1, 0, -1);
+    igraph_small(&g_2, 2, 0, -1);
+    igraph_small(&g_2c, 2, 0, 0,1, -1);
+    igraph_small(&g_wiki, 6, 1, 0,1, 0,2, 1,3, 2,1, 2,3, 2,4, 3,4, 3,5, 4,5, -1);
+    igraph_small(&g_wiki_u, 6, 0, 0,1, 0,2, 1,3, 2,1, 2,3, 2,4, 3,4, 3,5, 4,5, -1);
```

Comment:
```suggestion
    igraph_small(&g_0, 0, IGRAPH_UNDIRECTED, -1);
    igraph_small(&g_1, 1, IGRAPH_UNDIRECTED, -1);
    igraph_small(&g_2, 2, IGRAPH_UNDIRECTED, -1);
    igraph_small(&g_2c, 2, IGRAPH_UNDIRECTED, 0,1, -1);
    igraph_small(&g_wiki, 6, IGRAPH_DIRECTED, 0,1, 0,2, 1,3, 2,1, 2,3, 2,4, 3,4, 3,5, 4,5, -1);
    igraph_small(&g_wiki_u, 6, IGRAPH_UNDIRECTED, 0,1, 0,2, 1,3, 2,1, 2,3, 2,4, 3,4, 3,5, 4,5, -1);
```

Just for my sanity, there were too many numbers.